### PR TITLE
feat: SegmentedControl and Range components

### LINKED ISSUE #222: Add Segmented control component
Added in v1.11 => https://www.systeme-de-design.gouv.fr/elements-d-interface/composants/controle-segmente

### LINKED ISSUE #223: Add Range component
Added in v1.11 => https://www.systeme-de-design.gouv.fr/elements-d-interface/composants/curseur-range

I think we should implement it as a new type of the `Input` component.

### PR DESCRIPTION
**Pull request by lsagetlethias**
Closes #222
Closes #223

https://main--ds-gouv.netlify.app/example/component/segmented/
https://main--ds-gouv.netlify.app/example/component/range/

### CODE REVIEW

**Comment by lsagetlethias** [comment]
Range component is acting weird and freezing my tab even on dsfr netflify demo app.

I think there is a memory leak on dsfr.js but I can't figure it out where precisely.

**Comment by lsagetlethias** [comment]
Example on the DSFR website:
image

**Comment by lsagetlethias** [comment]
issue created on dsfr side: https://github.com/GouvernementFR/dsfr/issues/865

**Comment by garronej** [comment]
Le boss! Je regarde dès que je peut!

**Comment by lsagetlethias** [comment]
Ok top ! Attention toutefois, le Range est toujours buggé côté DSFR

File: src/Range.tsx [hunk 1]
```diff
@@ -0,0 +1,231 @@
+"use client";
+
+import React, {
+    CSSProperties,
+    DetailedHTMLProps,
+    forwardRef,
+    InputHTMLAttributes,
+    memo,
+    ReactNode
+} from "react";
```
Comment (garronej) [inline src/Range.tsx:10]:
```suggestion
import React, {
    type CSSProperties,
    forwardRef,
    memo,
    type ReactNode,
    type ComponentProps
} from "react";
```

File: src/Range.tsx [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    export type AsSingle = {
+        double?: never;
+        nativeInputProps?: NativeInputProps;
+    };
+
+    export type AsDouble = {
+        double: true;
+        nativeInputProps?: [NativeInputProps?, NativeInputProps?];
+    };
+}
```
Comment (garronej) [inline src/Range.tsx:73]:
```suggestion
export namespace RangeProps {

    export type AsSingle = {
        double?: never;
        nativeInputProps?: ComponentProps<"input">;
    };

    export type AsDouble = {
        double: true;
        nativeInputProps?: [ComponentProps<"input">?, ComponentProps<"input">?];
    };
}
```

File: src/SegmentedControl.tsx [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        hintText?: never;
+        hideLegend: true;
+    };
+
+    export type Segment = {
+        label: ReactNode;
+        nativeInputProps?: DetailedHTMLProps<
+            InputHTMLAttributes<HTMLInputElement>,
+            HTMLInputElement
+        >;
```
Comment (garronej) [inline src/SegmentedControl.tsx:70]: Same here, I think it's cleaner to use `React.ComponentProps<"input">` (I wasn't aware of this utility type before that's why it's not used in the other components 😄 )

**Review by garronej — APPROVED** [review]
Looks very good to me, Just a little suggestion improvement for expressiveness and consision and it can be merged!

**Pull request merged**